$NetBSD: patch-aa,v 1.28 2012/06/11 13:41:12 wiz Exp $

o Handle properly sh3 ports on NetBSD.
o Fix CPU family detection on NetBSD with /proc mounted with "-o linux"
  (please see also kern/25076).
o Fix a bug which broke the detection of a lot of X11 libraries.
o Add necessary libs for aalib support even if the aa option is turned on with
  --enable-aa.
o Add support for dvb capture on NetBSD.

--- configure.orig	2012-05-22 18:01:51.050593792 +0000
+++ configure
@@ -241,6 +241,7 @@ os2()       { issystem "OS/2"; }
 qnx()       { issystem "QNX"; }
 sunos()     { issystem "SunOS"; }
 wine()      { issystem "Wine"; }
+haiku()     { issystem "Haiku"; }
 win32()     { cygwin || mingw32 || wine; }
 
 # arch test boolean functions
@@ -486,6 +487,7 @@ Video output:
   --enable-vesa            enable VESA video output [autodetect]
   --enable-svga            enable SVGAlib video output [autodetect]
   --enable-sdl             enable SDL video output [autodetect]
+  --enable-haiku           enable Haiku-OS video and audio output [autodetect]
   --enable-kva             enable KVA video output [autodetect]
   --enable-aa              enable AAlib video output [autodetect]
   --enable-caca            enable CACA  video output [autodetect]
@@ -696,6 +698,7 @@ _xv=auto
 _xvmc=no  #auto when complete
 _vdpau=auto
 _sdl=auto
+_haiku=auto
 _kva=auto
 _direct3d=auto
 _directx=auto
@@ -1039,6 +1042,8 @@ for ac_option do
   --disable-vdpau)      _vdpau=no       ;;
   --enable-sdl)         _sdl=yes        ;;
   --disable-sdl)        _sdl=no         ;;
+  --enable-haiku)       _haiku=yes        ;;
+  --disable-haiku)      _haiku=no         ;;
   --enable-kva)         _kva=yes        ;;
   --disable-kva)        _kva=no         ;;
   --enable-direct3d)    _direct3d=yes   ;;
@@ -1485,7 +1490,7 @@ if test -z "$_target" ; then
   # OS name
   system_name=$(uname -s 2>&1)
   case "$system_name" in
-  Linux|FreeBSD|NetBSD|OpenBSD|DragonFly|BSD/OS|Darwin|SunOS|QNX|GNU|MorphOS|AIX|AmigaOS|Haiku)
+  Linux|FreeBSD|NetBSD|OpenBSD|DragonFly|BSD/OS|Darwin|SunOS|QNX|GNU|Haiku|MorphOS|AIX|AmigaOS|Haiku)
     ;;
   IRIX*)
     system_name=IRIX
@@ -1505,6 +1510,9 @@ if test -z "$_target" ; then
   OS/2*)
     system_name=OS/2
     ;;
+  Haiku)
+    system_name=Haiku
+    ;;
   *)
     system_name="$system_name-UNKNOWN"
     ;;
@@ -1513,14 +1521,15 @@ if test -z "$_target" ; then
 
   # host's CPU/instruction set
   case "$(uname -m 2>&1)" in
-      x86_64|amd64|i[3-9]86*|i86pc|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686) host_arch=i386 ;;
+      x86_64|amd64|i[3-9]86*|i86pc|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686|BePC) host_arch=i386 ;;
       ia64) host_arch=ia64 ;;
       macppc|ppc*|Power*) host_arch=ppc ;;
       alpha) host_arch=alpha ;;
       sun4*|sparc*) host_arch=sparc ;;
       parisc*|hppa*|9000*) host_arch=hppa ;;
       arm*|zaurus|cats) host_arch=arm ;;
-      sh3|sh4|sh4a) host_arch=sh ;;
+      sh3*) host_arch=sh ;;
+	  sh4|sh4a) host_arch=sh4 ;;
       s390) host_arch=s390 ;;
       s390x) host_arch=s390x ;;
       *mips*) host_arch=mips ;;
@@ -1544,6 +1553,7 @@ else # if test -z "$_target"
       morphos) system_name=MorphOS ;;
       amigaos) system_name=AmigaOS ;;
       mingw32*) system_name=MINGW32 ;;
+      haiku) system_name=Haiku ;;
       wine) system_name=Wine ;;
     esac
   done
@@ -1555,7 +1565,11 @@ else # if test -z "$_target"
 fi
 
 extra_cflags="-I. -Iffmpeg $extra_cflags"
-extra_ldflags="-lm $extra_ldflags"
+if haiku ; then
+	extra_ldflags="$extra_ldflags -lroot -lbe -lmedia -lgame -lsupc++"
+else
+	extra_ldflags="-lm $extra_ldflags"
+fi
 _timer=timer-linux.c
 _getch=getch2.c
 
@@ -1674,7 +1688,7 @@ else
       cc_name=$cc_name_tmp
       echocheck "$_cc version"
       cc_vendor=gnu
-      cc_version=$($_cc -dumpversion 2>&1)
+      cc_version=$( ( $_cc -dumpversion | sed -e 's/-nb[0-9]//' ) 2>&1 )
       case $cc_version in
         2.96*)
           cc_fail=yes
@@ -1798,7 +1812,7 @@ if x86 ; then
   # gather more CPU information
   pname=$($_cpuinfo | grep 'model name' | cut -d ':' -f 2 | head -n 1)
   pvendor=$($_cpuinfo | grep 'vendor_id' | cut -d ':' -f 2  | cut -d ' ' -f 2 | head -n 1)
-  pfamily=$($_cpuinfo | grep 'cpu family' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
+  pfamily=$($_cpuinfo | grep -i 'cpu family' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
   pmodel=$($_cpuinfo | grep -v 'model name' | grep 'model' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
   pstepping=$($_cpuinfo | grep 'stepping' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
 
@@ -2274,7 +2288,12 @@ case "$host_arch" in
     test $_fast_clz = "auto" && _fast_clz=yes
     ;;
 
-  sh|sh4)
+  sh)
+    arch='sh'
+    iproc='sh'
+    ;;
+
+  sh4)
     arch='sh4'
     iproc='sh4'
     ;;
@@ -2493,6 +2512,10 @@ EOF
     arch='generic'
     ;;
 
+  haiku)
+    arch='i586'
+    ;;
+    
   *)
     echo "The architecture of your CPU ($host_arch) is not supported by this configure script"
     echo "It seems nobody has ported MPlayer to your OS or CPU type yet."
@@ -3175,7 +3198,7 @@ cat > $TMPC << EOF
 int main(void) { gethostbyname(0); socket(AF_INET, SOCK_STREAM, 0); return 0; }
 EOF
 _socklib=no
-for ld_tmp in "" "-lsocket -ldnet" "-lsocket -lnsl" "-lnsl" "-lsocket" ; do
+for ld_tmp in "" "-lsocket -ldnet" "-lsocket -lnsl" "-lnsl" "-lsocket" "-lnetwork" ; do
   cc_check $ld_tmp && ld_sock="$ld_tmp" && _socklib=yes && break
 done
 test $_socklib = yes && test $_winsock2_h = auto && _winsock2_h=no
@@ -3600,7 +3623,7 @@ def_os2threads='#define HAVE_OS2THREADS 
 echocheck "pthread"
 if linux ; then
   THREAD_CFLAGS=-D_REENTRANT
-elif freebsd || netbsd || openbsd || bsdos ; then
+elif freebsd || netbsd || openbsd || bsdos || haiku ; then
   THREAD_CFLAGS=-D_THREAD_SAFE
 fi
 if test "$_pthreads" = auto ; then
@@ -3611,7 +3634,7 @@ int main(void) { pthread_t tid; return p
 EOF
 _pthreads=no
 if ! hpux ; then
-  for ld_tmp in "-lpthreadGC2" "" "-lpthread" "-pthread" ; do
+  for ld_tmp in "-lpthreadGC2" "" "-lpthread" "-pthread" "-lroot" ; do
     # for crosscompilation, we cannot execute the program, be happy if we can link statically
     cc_check $THREAD_CFLAGS $ld_tmp && (tmp_run || test "$ld_static") && ld_pthread="$ld_tmp" && _pthreads=yes && break
   done
@@ -4288,7 +4311,11 @@ if test "$_x11" = auto && test "$_x11_he
            -L/usr/X11/lib -L/usr/lib32 -L/usr/openwin/lib -L/usr/local/lib64 -L/usr/X11R6/lib64 \
            -L/usr/lib ; do
     if netbsd; then
-      ld_tmp="$I -lXext -lX11 $ld_pthread -Wl,-R$(echo $I | sed s/^-L//)"
+	  if test -n "$I" ; then
+        ld_tmp="$I -lXext -lX11 $ld_pthread -Wl,-R$(echo $I | sed s/^-L//)"
+	  else
+        ld_tmp="-lXext -lX11 $ld_pthread"
+	  fi
     else
       ld_tmp="$I -lXext -lX11 $ld_pthread"
     fi
@@ -4669,6 +4696,8 @@ if test "$_aa" = yes ; then
   def_aa='#define CONFIG_AA 1'
   if cygwin ; then
     libs_mplayer="$libs_mplayer $(aalib-config --libs | cut -d " " -f 2,5,6)"
+  else
+    libs_mplayer="$libs_mplayer $(aalib-config --libs)"
   fi
   vomodules="aa $vomodules"
 else
@@ -4756,7 +4785,6 @@ EOF
       extra_cflags="$extra_cflags $inc_tmp" && break
   done
 fi
-echores "$_dvb"
 if test "$_dvb" = yes ; then
   _dvbin=yes
   inputmodules="dvb $inputmodules"
@@ -4765,14 +4793,29 @@ if test "$_dvb" = yes ; then
   aomodules="mpegpes(dvb) $aomodules"
   vomodules="mpegpes(dvb) $vomodules"
 else
-  _dvbin=no
-  noinputmodules="dvb $noinputmodules"
-  def_dvb='#undef CONFIG_DVB'
-  def_dvbin='#undef CONFIG_DVBIN '
-  aomodules="mpegpes(file) $aomodules"
-  vomodules="mpegpes(file) $vomodules"
+  cat >$TMPC <<EOF
+#include <dev/dtv/dtvio.h>
+int main(void) {return 0;}
+EOF
+  cc_check && _dvb=yes
+  if test "$_dvb" = yes ; then
+    _dvbin=yes
+    inputmodules="dvb $inputmodules"
+    def_dvb='#undef CONFIG_DVB'
+    def_dvbin='#define CONFIG_DVBIN 1'
+    aomodules="mpegpes(file) $aomodules"
+    vomodules="mpegpes(file) $vomodules"
+  else
+    _dvbin=no
+    noinputmodules="dvb $noinputmodules"
+    def_dvb='#undef CONFIG_DVB'
+    def_dvbin='#undef CONFIG_DVBIN '
+    aomodules="mpegpes(file) $aomodules"
+    vomodules="mpegpes(file) $vomodules"
+  fi
 fi
 
+echores "$_dvb"
 
 if darwin; then
 
@@ -5393,6 +5436,29 @@ fi
 echores "$_v4l2"
 
 
+echocheck "Haiku-OS"
+if test "$_haiku" = auto || test "$_haiku" = yes ; then
+  cat > $TMPC << EOF
+int main(void) {
+#ifdef __HAIKU__
+  return 0;
+#endif
+}
+EOF
+   _haiku=no
+   cc_check && _haiku=yes
+fi 
+if test "$_haiku" = yes ; then
+  def_haiku='#define CONFIG_HAIKU 1'
+  vomodules="haiku $vomodules"
+  aomodules="haiku $aomodules"
+else
+  def_haiku='#undef CONFIG_HAIKU'
+  novomodules="haiku $novomodules"
+  noaomodules="haiku $noaomodules"
+fi
+echores "$_haiku"
+
 
 #########
 # AUDIO #
@@ -7639,7 +7705,7 @@ fi
 # (FIXME: 'echocheck "dynamic linking"' above and modify here accordingly)
 ld_dl_dynamic=''
 freebsd || netbsd || openbsd || dragonfly || bsdos && ld_dl_dynamic='-rdynamic'
-if test "$_real" = yes || test "$_xanim" = yes && ! win32 && ! qnx && ! darwin && ! os2 && ! sunos; then
+if test "$_real" = yes || test "$_xanim" = yes && ! win32 && ! qnx && ! darwin && ! os2 && ! sunos && ! haiku; then
   ld_dl_dynamic='-rdynamic'
 fi
 
@@ -8150,6 +8216,7 @@ RADIO_CAPTURE=$_radio_capture
 REAL_CODECS = $_real
 S3FB = $_s3fb
 SDL = $_sdl
+HAIKU = $_haiku
 SPEEX = $_speex
 STREAM_CACHE = $_stream_cache
 SGIAUDIO = $_sgiaudio
@@ -8647,6 +8714,7 @@ $def_quartz
 $def_s3fb
 $def_sdl
 $def_sdl_sdl_h
+$def_haiku
 $def_svga
 $def_tdfxfb
 $def_tdfxvid
